src/ucp/core/ucp_ep.c

```diff
@@ -215,18 +215,40 @@ static void ucp_ep_destory_uct_eps(ucp_ep_h ep)
     }
 }
 
-void ucp_ep_destroy(ucp_ep_h ep)
+ucs_status_ptr_t ucp_ep_disconnect_nb(ucp_ep_h ep)
 {
     ucp_worker_h worker = ep->worker;
 
-    ucs_debug("destroy ep %p", ep);
+    ucs_debug("disconnect ep %p", ep);
 
     UCS_ASYNC_BLOCK(&worker->async);
     ucp_ep_delete_from_hash(ep);
     ucp_ep_destory_uct_eps(ep);
     UCS_ASYNC_UNBLOCK(&worker->async);
 
     ucs_free(ep);
+
+    return NULL;
```

Comment:
please add comment in the code - the code will be extended to actually implement non-blocking disconnect